### ISSUE TITLE
Provide (optional) detailed connection timings in request/response debugging and/or events

### ISSUE THREAD

**Issue by jschoudt** [artilleryio/artillery#665]
E.g. https://blog.risingstack.com/measuring-http-timings-node-js/

Understanding the time spent on various phases of connection/response can help diagnose performance issues more quickly.

**Comment by jschoudt** [comment]
I'm poking around with this for my own use and probably will submit a PR. Wanted to get an issue open in case I missed it and it's already implemented or there are issues around volume of data/events or low level nodejs request timing that I'm not aware of.

**Comment by hassy** [comment]
Yes, that would be great! [Request.js](https://github.com/request/request) can already expose that information via the `timings` property on the response object, we just need to read that and make it available to the user.

**Comment by jschoudt** [comment]
Yeah, I have that hacked into my local code already in debug (I really want it in events so I can get it into metrics via the plugin). It's not putting in the tls time, which is one thing I'm currently interested in. I'm digging into the Request.js code to see if I'm doing something wrong or if I need to add it there.
